### DISCUSSION TITLE
Feature Request for Frigate 14

### DISCUSSION THREAD

**Discussion by mdv0** [blakeblackshear/frigate#12676] [Ideas]
Feature Request:

1.Separate Tabs for Timeline, Events, and Snapshots:
Currently, selecting "History" defaults to the timeline, consuming time. Adding separate tabs for "Timeline," "Events," and "Snapshots" will streamline navigation, this would not have mattered with a few cam but as i am recording 21 cam it takes some time to navigate between timeline then event due to there being a ton of events.

2.Snapshot Tab:
Introduce a "Snapshot" tab to view already saved snapshots directly in the Frigate UI, a feature missing in the new version 14 but available in Home Assistant.

**Comment by NickM-27** [comment]
1.  I would argue this is much less efficient. We have received feedback from users that they like the Timeline / Events tab because there are cases where you want to use the timeline most of the time but events are helpful for jumping between your activities. With your suggestion as a user I would have to jump between entirely different pages and find where I was, reapply any filters, etc instead of just pressing one button, making it more clicks than the UI is now. 
![Screen Shot 2024-07-30 at 06 45 46 AM](https://github.com/user-attachments/assets/d0260e7d-2984-4911-be5f-b7bcc58e5648)
Perhaps you can describe more specifically your use case, and how separate tabs would save time vs having access to both views in the same page and only needing to press a single button to switch between the timeline and list.

2. This technically already exists with the Frigate+ tab, but the purpose is for submitting to Frigate+ not for review. snapshots only show a single point in time. In the review tab you can hover (desktop) or swipe right (mobile) to view the preview and you can grab the scrub bar at the bottom to scrub the preview. This is faster than watching the full video and gives way more information than a single snapshot. Other views and features are planned for future Frigate releases

**Reply by mdv0** [reply]
I am sorry i was not clear before what i meant was to add a tab for event and snapshot next to history 
![Capture](https://github.com/user-attachments/assets/2c321604-5278-4c2c-a5a9-cd9b6c5b395d) not talking of removing the current ui u describe. Because currently if just want to view event from a specific camera i either have to go to review then filter specific cam then events or go to specific cam then history then events which is time consuming in my case because first of al it takes time to load all events from review after which it allows to filter specific cam.

**Reply by NickM-27** [reply]
>  Because currently if just want to view event from a specific camera i either have to go to review then filter specific cam then events or go to sopecific cam then history then events.

that is incorrect, the events / timeline view on the right only shows the activity from the camera that is currently selected. and you select another camera in the row at the bottom to switch to that camera and then only that cameras activity shows on the timeline / events on the right

**Reply by mdv0** [reply]
There seems to be a misunderstanding. My point is that when I want to review events from a specific camera, I have to go to the history section, which by default takes me to the timeline view. I am suggesting an option to toggle the default view, so users can choose to go directly to the events view instead of the timeline if they prefer.

**Reply by NickM-27** [reply]
Right, but that is saving one click in the total time that the UI is viewed which is very inconsequential. The way it is described in the OP makes it seem like you are having to switch to the events view from timeline very often which indicates that the UI may be being used incorrectly due to some misunderstanding of how the UI works

**Reply by mdv0** [reply]
I am sorry for that as u might have probably noticed that english is not my first language and that is why u might have misunderstood my original request. maybe i am using ui wong, but i am only asking for this because for some reason the review tab takes much more time in frigate 14 than it used to in frigate 13 which was called events previously, it takes a good 5-10 sec to load events and then i can use the ui and us filtere or choose specific cam same is when i go to history it takes time to goto events from timeline. Maybe there is something worng with my setup,

**Reply by NickM-27** [reply]
Oh okay I understand what you are saying. Yes that does seem a bit odd, I've never seen any delay switching between these and we haven't gotten that feedback. I'd suggest using the network tab of the browser tools to see what happens when you switch back and forth, maybe something is slow to return causing the delay

**Reply by mdv0** [reply]
I have a better idea. How about you don't change any of the above-discussed UI tabs, and instead, add an option to remember the last filters or which cameras were previously selected in the reviews? If someone doesn't want this, just add a toggle for whether you want to save/remember the last settings/filters and selected camera or grouo/layout in reviews. Just like in 14-rc1, where you have added a toggle to view video or make the video bar static with only the image. In short there should be toggle which would allow to save last state of selected cam, filter etc in review section. And for the second suggestion related to snapshot, if in your opinion reviews is more useful then why not just integrate snapshot within timeline view like this 
![Screenshot_2024-07-31-22-48-25-352_com android chrome-edit](https://github.com/user-attachments/assets/e1956373-ee6e-4837-af3a-0d063613cc82).  And just to be clear plz dont mis judge my intension is solely to help improve frigate. Thanks
